src/hooks/useMaker.js

```diff
@@ -38,13 +38,35 @@ function useMaker() {
           'browser ethereum provider and URL network param do not match.'
         );
 
-      if (!isConnectedToProvider(browserProvider))
-        await maker.addAccount({
+      if (!browserProvider.address.match(/^0x[a-fA-F0-9]{40}$/))
+        throw new Error(
+          'browser ethereum provider providing incorrect or non-existent address'
+        );
+
+      let metaMaskAccount;
+      if (!maker.service('accounts').hasAccount) {
```

Comment:
tiny style thing - sometimes negative conditionals are ever so slightly harder to parse. this is a good reference https://github.com/ryanmcdermott/clean-code-javascript#avoid-negative-conditionals

Reply:
I can change it. My thinking there was that as the negative case's action is straightforward it would be better get it out of the way.

Reply:
I agree with @jparklev but I do see your reasoning there @sirromdev. The case in which using the negative conditional definitely makes sense (at least to my taste) is when you can `return` in the if-clause and then not need an `else` at all.